### PR TITLE
feat: update packages

### PR DESCRIPTION
**Pull request by XVincentX**
This will help to dedupe dependencies around

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
     "build": "sl-scripts build",
     "build.docs": "sl-scripts build:typedoc",
     "commit": "git-cz",
-    "lint": "tslint 'src/**/*.ts'",
+    "lint": "tslint -p tsconfig.json 'src/**/*.ts'",
```
Comment (P0lip) [inline package.json]: Why is this required?

File: package.json [hunk 2]
```diff
@@ -46,28 +46,28 @@
     "test.watch": "yarn test --watch"
   },
   "dependencies": {
-    "@stoplight/json": "1.9.x",
+    "@stoplight/json": "2.1.0",
```
Comment (P0lip) [inline package.json]:
```suggestion
    "@stoplight/json": "^2.1.0",
```

File: package.json [hunk 3]
```diff
@@ -46,28 +46,28 @@
...
     "dependency-graph": "0.8.x",
     "fast-memoize": "2.x.x",
-    "immer": "2.x.x",
+    "immer": "3.1.2",
```
Comment (P0lip) [inline package.json]:
```suggestion
    "immer": "^3.1.2",
```

File: src/runner.ts [hunk 4]
```diff
@@ -148,7 +148,7 @@ export class ResolveRunner implements Types.IResolveRunner {
 
         if (!r.resolved.result) continue;
 
-        this._source = produce(this._source, draft => {
+        this._source = produce(this._source, (draft: any) => {
```
Comment (P0lip) [inline src/runner.ts]: Vincenzo resorting to any :scream:

Comment (XVincentX) [reply src/runner.ts]: It was any even before, but now with Immer 3.0 is not wrapped anymore in stuff. At least, we can now shame ourselves.

File: package.json (hunk 1 above)
Comment (XVincentX) [reply package.json]: By giving the tsconfig file tslint has types informations which some rules require — such as `no-floating-promises`

Comment (P0lip) [reply package.json]: Ah, makes sense. I thought it would just default to it. Nice.

Comment (XVincentX) [reply package.json]: I thought the same to be honest!
